src/hotspot/share/runtime/lockStack.inline.hpp

```diff
@@ -42,8 +46,8 @@ JavaThread* LockStack::get_thread() const {
   return reinterpret_cast<JavaThread*>(addr - lock_stack_offset);
 }
 
-inline bool LockStack::can_push() const {
-  return to_index(_top) < CAPACITY;
+inline bool LockStack::is_full() const {
+  return to_index(_top) == CAPACITY;
```

Comment:
Would it be too paranoid to use `>= CAPACITY`?
Or to add an assert that the index is not greater than capacity?

Reply:
Added an assert to `to_index` that the offset is valid, also added verify calls to the `_recursive` functions.